src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/DefaultRazorFormattingService.cs

```diff
@@ -71,7 +76,14 @@ public override async Task<TextEdit[]> FormatAsync(DocumentUri uri, DocumentSnap
             return filteredEdits;
         }
 
-        public override async Task<TextEdit[]> ApplyFormattedEditsAsync(DocumentUri uri, DocumentSnapshot documentSnapshot, RazorLanguageKind kind, TextEdit[] formattedEdits, FormattingOptions options, CancellationToken cancellationToken)
+        public override async Task<TextEdit[]> ApplyFormattedEditsAsync(
+            DocumentUri uri,
+            DocumentSnapshot documentSnapshot,
+            RazorLanguageKind kind,
+            TextEdit[] formattedEdits,
+            FormattingOptions options,
+            CancellationToken cancellationToken,
+            bool bypassValidationPasses = false)
```

Comment:
Why do we want to bypass validation?

Reply:
The IFormatPass validations are mainly to ensure the formatting doesn't touch anything but whitespace. We enforce this behavior for on-type formatting / auto-formatting, but that of course wouldn't be something we want to enforce for code actions. @ajaybhargavb do I have that right?

Reply:
Yup. CodeActions typically change both content and structure which will be rejected by our validation passes. The other alternative here was to apply the code action edits as is before invoking the formatting pipeline. In this case, we will have to reparse and create a new DocumentSnapshot and then pass that into the formatting service.